pkg/serviceaccount/legacy.go

```diff
@@ -96,6 +107,27 @@ func (v *legacyValidator) Validate(ctx context.Context, tokenData string, public
 		return nil, errors.New("sub claim is invalid")
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.LegacyServiceAccountTokenTracking) {
+		warning.AddWarning(ctx, "", "Legacy token was used. Consider switching to [TokenRequest](https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/token-request-v1/) API.")
```

Comment:
A few notes:
* Markdown won't work in these warnings, use plain text
* This should be shorter, limit to 120 characters if possible
* Only warn when an auto-generated token is used (secret name is in the serviceAccount.Secrets list)

Something like this might work better:

> Use tokens from the TokenRequest API or dedicated secret-based tokens instead of auto-generated service account tokens.

If you want to link to a page with docs (which is probably a good idea), make sure there's a single doc destination and add a short redirect to it to something like `https://k8s.io/legacy-tokens` (defined in https://github.com/kubernetes/website/blob/main/static/_redirects)

Reply:
done